src/use-built-ins-plugin.js

```diff
@@ -0,0 +1,227 @@
+      t.isIdentifier(path.node.expression.callee) &&
+      path.node.expression.callee.name === "require" &&
+      path.node.expression.arguments.length === 1 &&
+      t.isStringLiteral(path.node.expression.arguments[0]) &&
+      isPolyfillSource(path.node.expression.arguments[0].value);
+  }
+
+  const addAndRemovePolyfillImports = {
+    ImportDeclaration(path) {
+      if (
+        path.node.specifiers.length === 0 &&
+        isPolyfillSource(path.node.source.value)
+      ) {
+        console.warn(
+          `
+Adding "import 'babel-polyfill'" isn't necessary with the useBuiltIns option anymore.
+Please remove the call.
+`,
+        );
+        path.remove();
+      }
+    },
+    Program: {
+      enter(path, state) {
+        if (!state.opts.polyfills) {
+          throw path.buildCodeFrameError(
+            `
+There was an issue in "babel-preset-env" such that
+the "polyfills" option was not correctly passed
+to the "transform-polyfill-require" plugin
+`,
+          );
+        }
+        path.get("body").forEach(bodyPath => {
+          if (isRequire(bodyPath)) {
+            console.warn(
+              `
+Adding "require('babel-polyfill')" isn't necessary with the useBuiltIns option anymore.
+Please remove the call.
+`,
+            );
+            path.remove();
+          }
+        });
+      },
+    },
+
+    // Symbol() -> _core.Symbol();
+    // new Promise -> new _core.Promise
+    ReferencedIdentifier(path, state) {
+      const { node, parent, scope } = path;
+
+      if (t.isMemberExpression(parent)) return;
+      if (!has(definitions.builtins, node.name)) return;
+      if (scope.getBindingIdentifier(node.name)) return;
+
+      const builtIn = definitions.builtins[node.name];
+      addUnsupported(path, state.opts.polyfills, builtIn, this.builtIns);
+    },
+
+    // Array.from -> _core.Array.from
+    MemberExpression: {
+      enter(path, state) {
+        if (!path.isReferenced()) return;
+
+        const { node } = path;
+        const obj = node.object;
+        const prop = node.property;
+
+        if (!t.isReferenced(obj, node)) return;
+
+        // doesn't reference the global
+        if (path.scope.getBindingIdentifier(obj.name)) return;
+
+        if (has(definitions.staticMethods, obj.name)) {
+          const staticMethods = definitions.staticMethods[obj.name];
+          if (has(staticMethods, prop.name)) {
+            const builtIn = staticMethods[prop.name];
+            addUnsupported(path, state.opts.polyfills, builtIn, this.builtIns);
+          }
+        }
+
+        if (
+          !node.computed &&
+          t.isIdentifier(prop) &&
+          has(definitions.instanceMethods, prop.name)
+        ) {
+          state.opts.debug && warnOnInstanceMethod(getObjectString(node));
+          const builtIn = definitions.instanceMethods[prop.name];
+          addUnsupported(path, state.opts.polyfills, builtIn, this.builtIns);
+        } else if (
+          node.computed &&
```

Comment:
This is very cool that it catches: `b['find']` but it won't catch `var c = 'find'; b[c];`. 

And I know it's literally impossible to statically guarantee that you'll find all cases because of stuff like `var c = 'fi' + 'nd'; b[c];`

However, maybe it makes sense to iterate over all string literals and if any match an instance method name, include the polyfill?

I only say that because it's far more likely to see: `b[someCondition ? 'find' : 'findIndex']` and you never see the literal string form because you can use non computed *always* in that case.